### ISSUE TITLE
Shorthand destructuring defaults in arrow function params

### ISSUE THREAD

**Issue by johanneslumpe** [6to5/6to5#576]
```javascript
({some = 'test', another = true} = {}) => {}
```

The above fails to compile in the REPL with the following error:
````
javascript
repl: Unexpected token (1:7)
> 1  | ({some = 'test', another = true} = {}) => {}
     |       ^
```

But using the Traceur transcoding demo, it compiles fine to:

````javascript
$traceurRuntime.ModuleStore.getAnonymousModule(function() {
  "use strict";
  (function() {
    var $__1,
        $__2;
    var $__0 = arguments[0] !== (void 0) ? arguments[0] : {},
        some = ($__1 = $__0.some) === void 0 ? 'test' : $__1,
        another = ($__2 = $__0.another) === void 0 ? true : $__2;
  });
  return {};
});
//# sourceURL=traceured.js
```

**Comment by sebmck** [comment]
```javascript
({some: some = 'test', another: another = true} = {}) => {}
```

is supported but shorthand defaults aren't currently for arrow functions. See marijnh/acorn#181

**Comment by johanneslumpe** [comment]
Ah sorry, didn't see the other issue!

**Comment by sebmck** [comment]
Now supported upstream via the amazing work of @RReverser  marijnh/acorn@65d09eac6eeeec5dda2a03db715e881892a9675d I'll need to fix up quite a few merge conflicts as I've made some rather stupid modifications to acorn in order to get async functions to parse correctly.

**Comment by johanneslumpe** [comment]
:+1:

**Comment by sebmck** [comment]
Merged and will be available in 3.0.0
